### PR TITLE
2-add-database-functions-that-are-needed-for-basic-gui

### PR DESCRIPTION
**Pull request by soerenlemke**

### CODE REVIEW

File: database/getters.go [hunk 1]
```diff
@@ -33,3 +41,33 @@ func (g *Get) AllTableNames() ([]string, error) {
 
 	return tableNames, nil
 }
+
+func (g *Get) TableColumns(t string) ([]ColumnInfo, error) {
+	queryString := fmt.Sprintf("PRAGMA table_info(%s);", t)
```
Comment (vyshnav-vinod) [inline database/getters.go:46]: I belive we have to use [Prepare](https://pkg.go.dev/database/sql#DB.Prepare) for passing variables. This looks good for now, but we should look into this sooner or later

File: database/types.go [hunk 2]
```diff
@@ -16,3 +16,17 @@ type Get struct {
 type Set struct {
 	DB *Database
 }
+
+type ColumnInfo struct {
+	Name string
+	Type string
+}
+
+type TableColumn struct {
```
Comment (vyshnav-vinod) [inline database/types.go:25]: Can you add comments here? Also ColumnInfo seems redundant here. We already have name and type in TableColumn. Or is it for something else? You can use comments here to specify what each struct will be used for.

File: database/getters_test.go [hunk 3]
```diff
@@ -63,3 +62,36 @@ func TestGet_DatabaseName(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			g := &Get{
+				DB: tt.fields.DB,
+			}
+			got, err := g.TableColumns(tt.args.t)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("TableColumns() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```
Comment (vyshnav-vinod) [inline database/getters_test.go:92]: The compiler throws warning here stating not to use DeepEqual to compare errors. I will work on the testing after we merge this.

File: database/getters.go [hunk 4]
```diff
@@ -20,6 +22,12 @@ func (g *Get) AllTableNames() ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer func(rows *sql.Rows) {
+		err := rows.Close()
+		if err != nil {
+			log.Println("Error closing rows:", err)
```
Comment (vyshnav-vinod) [inline database/getters.go:28]: Should we log this? Or should we return as err? We cant `return nil, err` here as we already got the table names value. What do you think will be a good approach to this.

**Review by vyshnav-vinod — COMMENTED** [review]
Most of it looks good. But i have left some comments.

File: database/getters.go (hunk 1 above)
Comment (soerenlemke) [reply database/getters.go:46]: added it to the code

File: database/types.go (hunk 2 above)
Comment (soerenlemke) [reply database/types.go:25]: You`re right! Don`t know why i have both. Refactored it to only use TableColumn.

File: database/getters_test.go (hunk 3 above)
Comment (soerenlemke) [reply database/getters_test.go:92]: ok sounds good.

File: database/getters.go (hunk 4 above)
Comment (soerenlemke) [reply database/getters.go:28]: I added logging for now. Im not quite sure what is happening behind the scenes when we get an error closing it and keep going on.

Comment (vyshnav-vinod) [reply database/getters.go:28]: I guess it doesnt matter. [docs[(https://pkg.go.dev/database/sql#Rows.Close)

**Pull request closed without merging**